### PR TITLE
Add a functional test for podman

### PR DESCRIPTION
**Pull request by ilya-zuyev**
Fixes https://github.com/kubernetes/minikube/issues/10673

`TestMountCmd` is failing and is skipped for now for podman/GitHub env, see https://github.com/kubernetes/minikube/issues/11293

### CODE REVIEW

File: test/integration/functional_test.go [hunk 1]
```diff
@@ -416,6 +413,63 @@ func validateDockerEnv(ctx context.Context, t *testing.T, profile string) {
...
+	if runtime.GOOS != "linux" {
+		t.Skipf("only validate podman env on linux, currently testing %s", runtime.GOOS)
+	}
+
+	defer PostMortemLogs(t, profile)
+
+	mctx, cancel := context.WithTimeout(ctx, Seconds(120))
+	defer cancel()
+
+	c := exec.CommandContext(mctx, "/bin/bash", "-c", "eval $("+Target()+" -p "+profile+" docker-env) && "+Target()+" status -p "+profile)
```
Comment (medyagh) [inline test/integration/functional_test.go:435]: I think u meant podman-env here

**Comment by medyagh** [comment]
seems to be also trying to fix https://github.com/kubernetes/minikube/issues/10231

File: test/integration/functional_test.go (hunk 1 above)
Comment (ilya-zuyev) [reply test/integration/functional_test.go]: Fixed. Thanks!

File: test/integration/functional_test.go [hunk 2]
```diff
@@ -460,6 +459,61 @@ func validateDockerEnv(ctx context.Context, t *testing.T, profile string) {
 	}
 }
 
+// check functionality of minikube after evaling docker-env
```
Comment (medyagh) [inline test/integration/functional_test.go:462]: fix comment

File: .github/workflows/pr.yml [hunk 3]
```diff
@@ -280,6 +279,109 @@ jobs:
           if [ "$numPass" -eq 0 ];then echo "*** 0 Passed! ***";exit 2;fi
           if [ "$numPass" -lt 36 ];then echo "*** Failed to pass at least 36 ! ***";exit 2;fi
           if [ "$numPass" -eq 0 ];then echo "*** Passed! ***";exit 0;fi
+
+  functional_podman_ubuntu:
```
Comment (medyagh) [inline .github/workflows/pr.yml:283]: add same thing to master.yml

File: Makefile [hunk 4]
```diff
@@ -948,3 +948,4 @@ cpu-benchmark-autopause: ## run the cpu usage auto-pause benchmark
 update-gopogh-version: ## update gopogh version
 	(cd hack/update/gopogh_version && \
 	 go run update_gopogh_version.go)
+
```
Comment (medyagh) [inline Makefile:951]: remove unrelated

also please squash the commits, for a cleaner commmit history

**Review by medyagh — CHANGES_REQUESTED** [review]

**Comment by ilya-zuyev** [comment]
recreated as https://github.com/kubernetes/minikube/pull/11295

**Pull request closed without merging**